XenAdmin/Wizards/PatchingWizard/PatchingWizard_SelectServers.cs

```diff
@@ -96,7 +96,9 @@ protected override void PageLoadedCore(PageLoadedDirection direction)
         {
             try
             {
-                poolSelectionOnly = WizardMode == WizardMode.AutomatedUpdates || SelectedUpdateAlert != null || FileFromDiskAlert != null;
+                List<IXenConnection> xenConnections = ConnectionsManager.XenConnectionsCopy;
+                bool hasMixedPool = xenConnections.Any(xc => Helpers.GetPool(xc) != null && !Helpers.GetPool(xc).IsPoolFullyUpgraded());

```

Comment:
Trying to avoid calling `GetPool` twice, this can be written as:
`hasMixedPool = xenConnections.Select(Helpers.GetPool).Any(p => p != null && !p.IsPoolFullyUpgraded());`